### PR TITLE
Initial cloud foundry changes

### PR DESCRIPTION
**Pull request by AndrewTorrance**
## Checklist
  - CHANGELOG.md updated?

### CODE REVIEW

**Comment by AndrewTorrance** [comment]
[2nd 1/2 SDX into Cloudfoundry (5)](https://trello.com/c/bB87Q2Se/522-2nd-1-2-sdx-into-cloudfoundry-5)

File: manifest.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+- name: sdx-downstream
+  buildpack: python_buildpack
+  health-check-type: process
+
+  env:
+    CF_DEPLOYMENT: True
+    SDX_STORE_URL: http://sdx-store:5000"
+    SDX_TRANSFORM_CS_URL: http://sdx-transform-cs:5000
+    SDX_TRANSFORM_CORA_URL: http://sdx-transform-cora:5000
+    SDX_SEQUENCE_URL:  http://sdx-sequence:5000
```
Comment (gavinleeedwards) [inline manifest.yml]: Should these relate to the CF URLs (http://sdx-store.apps.devtest.onsclofo.uk)?

File: app/settings.py [hunk 2]
```diff
@@ -1,46 +1,60 @@
...
+if os.getenv("CF_DEPLOYMENT", False):
+    RABBIT_URL, RABBIT_URL2, FTP_USER, FTP_PASS = parse_vcap_services()
+else:
+    # Default to true, cast to boolean
+    SDX_STORE_URL = os.getenv("SDX_STORE_URL", "http://sdx-store:5000")
+    SDX_TRANSFORM_CS_URL = os.getenv("SDX_TRANSFORM_CS_URL", "http://sdx-transform-cs:5000")
+    SDX_TRANSFORM_CORA_URL = os.getenv("SDX_TRANSFORM_CORA_URL", "http://sdx-transform-cora:5000")
+    SDX_SEQUENCE_URL = os.getenv("SDX_SEQUENCE_URL", "http://sdx-sequence:5000")
+
+    FTP_HOST = os.getenv('FTP_HOST', 'pure-ftpd')
```
Comment (gavinleeedwards) [inline app/settings.py]: Where does the FTP host come from if it's a CF deployment?  Is FTP an available CF service for us to use?

File: manifest.yml (hunk 1 above)
Comment (AndrewTorrance) [reply manifest.yml]: Your right . I thinkI have done the same on several repos- will revisit

File: app/settings.py [hunk 3]
```diff
@@ -1,46 +1,60 @@
...
+    parsed_vcap_services = json.loads(vcap_services)
+    rabbit_config = parsed_vcap_services.get('rabbitmq')
+    rabbit_url = rabbit_config[0].get('credentials').get('uri')
+    rabbit_url2 = rabbit_config[1].get('credentials').get('uri') if len(rabbit_config) > 1 else rabbit_url
+
+    ftp_config = parsed_vcap_services.get('ftp')
+    ftp_user = ftp_config[0].get('ftp_user')
+    ftp_pass = ftp_config[0].get('ftp_pass')
+
+    return rabbit_url, rabbit_url2, ftp_user, ftp_pass
```
Comment (warrenbailey) [inline app/settings.py]: I added a unit test for this in collect should help to stop the code coverage reducing:
https://github.com/ONSdigital/sdx-collect/blob/master/tests/test_settings.py

**Pull request merged**